3-extra/2-array-of-objects.js

```diff
@@ -11,7 +11,30 @@
 */
 
 function getHighestRatedInEachGenre(books) {
-    // TODO
+    const theHighest = [], theGenres = [], theTitles = [];
+    let theIndex;
+    for (const eachBook of books) { // go through each book
+        // has this 'genre' been seen before - search for it in 'theGenres'
```

Comment:
@DelroyGayle  why it is important that the genre has seen before or not. the important thing here is the rating. you can think about the rating instead.
I think you will learn more about objects next session, if you want to leave this and come back to it maybe it will be easier.